extensions/amp-analytics/0.1/visibility-model.js

```diff
@@ -346,13 +298,16 @@ export class VisibilityModel {
     }
     const conditionsMet = this.updateCounters_(visibility);
     if (conditionsMet) {
-      if (this.scheduledRunId_) {
-        clearTimeout(this.scheduledRunId_);
-        this.scheduledRunId_ = null;
+      if (this.scheduledUpdateTimeoutId_) {
+        clearTimeout(this.scheduledUpdateTimeoutId_);
+        this.scheduledUpdateTimeoutId_ = null;
       }
       if (this.reportReady_) {
         this.eventResolver_();
         this.eventResolver_ = null;
+        if (this.repeat_) {
+          this.waitToReset_ = true;
```

Comment:
We currently do this on the next update (see line 292) but happy to test moving it down here and report back.

Reply:
that line is guarded by a `!this.isVisibilityMatch_` check. if you don't reset duration number here, I'm afraid that it will never pass the check.

you can test on your example to see if I'm right about this:
1) have a repeat trigger with continuousVisibleMin set to 2s. 
2) let it keep visible for 7s 
3) you should see 3 pings

Reply:
Ok I tested with copying these lines to just after line 309:
```
          this.continuousTime_ = 0;
          this.maxContinuousVisibleTime_ = 0;
```
I do not see any difference in behaviour, e.g. manual and unit tests passed both before and after the change.

Reply:
(Posted my last message before reading yours...)

Reply:
in fact, `!this.isVisibilityMatch_` is not even checking duration.
so now i'm very curious how the code works.

can you give a pointer to the real logic?

Reply:
Including maxContinuousVisibleTime_ above was incorrect. I've now added the "this.continuousTime_ = 0;"
To deduce prior logic, I added (did not check in, of course!) a log statement at the bottom of this method, to log the value of this.continuousTime_, and also log statements when reset_() is called. What was happening, which was imperceptibly close to correct, was that update_() would end with continuousTime_=2000, then in the next call, reset_() would fire on line 331, setting it to zero. So we were off by one update cycle. This is fixed now, I believe.